### PR TITLE
GUI: Add a renderer selection list

### PR DESCRIPTION
**Pull request by bgK**
This PR is another take at what was attempted in #820 and #848. It replaces the "Software renderer" checkbox in the options dialog with a "Renderer" list. This allows building simultaneously the legacy OpenGL and modern OpenGL renderers, and lets the user chose the renderer to use.

The modern OpenGL renderers are now built by default, but the legacy OpenGL renderers are still used by default at runtime.

A side effect is that the "soft_renderer" configuration option is not used anymore, breaking compatibility with existing configuration files.

### CODE REVIEW

**Comment by bgK** [comment]
@aquadran today's new version includes fixes for your comments

**Comment by aquadran** [comment]
@bgK please update README too

File: gui/options.cpp [hunk 1]
```diff
@@ -789,8 +794,17 @@ void OptionsDialog::addGraphicControls(GuiObject *boss, const Common::String &pr
 
 	// ResidualVM specific description
 	_aspectCheckbox = new CheckboxWidget(boss, prefix + "grAspectCheckbox", _("Preserve aspect ratio"), _("Preserve the aspect ratio in fullscreen mode"));
-// ResidualVM specific option:
-	_softwareRenderingCheckbox = new CheckboxWidget(boss, prefix + "grSoftwareRendering", _("Software Rendering"), _("Enable software rendering"));
+	// ResidualVM specific -- Start
+	_rendererTypePopUpDesc = new StaticTextWidget(boss, prefix + "grRendererTypePopupDesc", _("Game Renderer:"));
```
Comment (aquadran) [inline gui/options.cpp]: hmm, Game or Graphics Renderer?

**Comment by aquadran** [comment]
:+1:

**Comment by Botje** [comment]
:+1:

I'll be so happy once we can close those old PRs!

**Comment by bgK** [comment]
I will merge this tomorrow unless somebody wants to further review it.

**Pull request merged**